### PR TITLE
Formatting errors and warnings for bad formatting

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1290

## This PR contains the following changes

- {List all the relevant changes. If there are many changes across many Jira tickets, organize the changes by Jira ticket}

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [x] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

## Screenshots
Screen Shot 2021-08-31 at 12 04 03 PM
Screen Shot 2021-08-31 at 12 03 33 PM
Screen Shot 2021-08-31 at 8 54 34 AM

### CODE REVIEW

File: n8n/.n8n/config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (NickPhura) [inline n8n/.n8n/config]: This file can be removed.  Its probably a remnant from early n8n stuff, before we changed the temp folder name from `n8n` to `.n8n`

File: .n8n/config [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (NickPhura) [inline .n8n/config]: This file can be removed. Its probably a remnant from early n8n stuff, before we changed the temp folder name from n8n to .n8n

File: api/src/utils/media/csv/validation/csv-row-validator.ts [hunk 3]
```diff
@@ -37,6 +37,8 @@ export const getRequiredFieldsValidator = (requiredFieldsByHeader?: string[]): C
       for (const requiredFieldByHeader of requiredFieldsByHeader) {
         const columnIndex = headers.indexOf(requiredFieldByHeader);
 
+        //if column does not exist, return
```
Comment (sdevalapurkar) [inline api/src/utils/media/csv/validation/csv-row-validator.ts]: nit:
```suggestion
        //if column does not exist, return csv worksheet
```

File: api/src/utils/media/csv/validation/csv-row-validator.ts [hunk 4]
```diff
@@ -174,3 +194,52 @@ export const getValidRangeFieldsValidator = (requiredRangeByHeader?: IValueRange
     return csvWorksheet;
   };
 };
+
+/**
+ * For each item in `requiredFormatsByHeader`, adds an error for each row cell whose value does not match the regular expression pattern.
+ *
+ * Note: If the cell is empty, this check will be skipped.  Use the `getRequiredFieldsValidator` validator to assert
```
Comment (sdevalapurkar) [inline api/src/utils/media/csv/validation/csv-row-validator.ts]:
```suggestion
 * Note: If the cell is empty, this check will be skipped. Use the `getRequiredFieldsValidator` validator to assert
```

File: app/src/features/surveys/view/SurveyObservations.tsx [hunk 5]
```diff
@@ -323,12 +341,32 @@ const SurveyObservations = () => {
                 Resolve the following errors in your local file and re-import.
               </Typography>
             </Box>
+
+            <Box>
+              {Object.entries(submissionErrors).map(([key, value], index) => {
```
Comment (sdevalapurkar) [inline app/src/features/surveys/view/SurveyObservations.tsx]: can just do:
```suggestion
              {Object.entries(submissionErrors).map(([key, value], index) => (
```
and then remove the `return` statement

File: database/src/migrations/release.0.23/populate_submission_message_type.sql [hunk 6]
```diff
@@ -11,7 +11,6 @@ insert into submission_message_type (name, record_effective_date, description, s
 
 -- field related
 insert into submission_message_type (name, record_effective_date, description, submission_message_class_id) values ('Missing Required Field', now(), 'A required field was not detected.', (select submission_message_class_id from submission_message_class where name = 'Error'));
-insert into submission_message_type (name, record_effective_date, description, submission_message_class_id) values ('Unexpected Format', now(), 'A field with an unexpected format was detected.', (select submission_message_class_id from submission_message_class where name = 'Error'));
+insert into submission_message_type (name, record_effective_date, description, submission_message_class_id) values ('Unexpected Format', now(), 'A field with an unexpected format was detected.', (select submission_message_class_id from submission_message_class where name = 'Warning'));
```
Comment (NickPhura) [inline database/src/migrations/release.0.23/populate_submission_message_type.sql]: Why the change to warning?

I would assume a regex rule would be an error, otherwise why bother with it at all.  We could just leave it at being a string or something

File: app/src/features/surveys/view/SurveyObservations.tsx [hunk 7]
```diff
@@ -228,24 +235,38 @@ const SurveyObservations = () => {
...
 
   const messageList = submissionStatus?.messages;
 
   if (messageList) {
     Object.entries(messageGrouping).forEach(([key, value]) => {
       messageList.forEach((message) => {
-        if (value.type.includes(message.type)) {
-          if (!submissionMessages[key]) {
-            submissionMessages[key] = [];
+        if (value.type.includes(message.type) && message.class === ClassGrouping.ERROR) {
```
Comment (NickPhura) [inline app/src/features/surveys/view/SurveyObservations.tsx]: nit: can you combine the 2 if statements (for the value.type.includes portion)

File: database/src/migrations/release.0.23/populate_submission_message_type.sql (hunk 6 above)
Comment (anissa-agahchen) [reply database/src/migrations/release.0.23/populate_submission_message_type.sql]: we need to show something for the warning category ... should I duplicate is with a different name?

File: app/src/features/surveys/view/SurveyObservations.tsx (hunk 7 above)
Comment (anissa-agahchen) [reply app/src/features/surveys/view/SurveyObservations.tsx]: Oh right .. forgot

**Review by NickPhura — DISMISSED** [review]

**Review by NickPhura — APPROVED** [review]
